extensions/blocks/publicize/store/effects.js

```diff
@@ -1,32 +1,121 @@
 /**
  * External dependencies
  */
+import { flatMap, throttle } from 'lodash';
 import apiFetch from '@wordpress/api-fetch';
+import { serialize } from '@wordpress/blocks';
+import { select, dispatch } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
-import { setConnectionTestResults } from './actions';
+import { SUPPORTED_CONTAINER_BLOCKS } from '../twitter';
 
 /**
  * Effect handler which will refresh the connection test results.
  *
- * @param {Object} action Action which had initiated the effect handler.
- * @param {Object} store  Store instance.
- *
- * @return {Object} Refresh connection test results action.
+ * @returns {object} Refresh connection test results action.
  */
-export async function refreshConnectionTestResults( action, store ) {
-	const { dispatch } = store;
-
+export async function refreshConnectionTestResults() {
 	try {
 		const results = await apiFetch( { path: '/wpcom/v2/publicize/connection-test-results' } );
-		return dispatch( setConnectionTestResults( results ) );
+		return dispatch( 'jetpack/publicize' ).setConnectionTestResults( results );
 	} catch ( error ) {
 		// Refreshing connections failed
 	}
 }
 
+/**
+ * Given an array of blocks, this will return an array of just the blocks (including child blocks of
+ * those blocks passed) that we support transforming into tweet content.
+ *
+ * @param {Array} blocks - The array of blocks to check.
+ * @returns {Array} The blocks that can be turned into tweets.
+ */
+export const computeTweetBlocks = ( blocks = [] ) => {
+	const { getSupportedBlockType } = select( 'jetpack/publicize' );
+	return flatMap( blocks, ( block = {} ) => {
+		if ( getSupportedBlockType( block.name ) ) {
+			return block;
+		}
+
+		// core-embed/* blocks are a special case.
+		if ( block.name.startsWith( 'core-embed/' ) ) {
+			return block;
+		}
+
+		if ( SUPPORTED_CONTAINER_BLOCKS.includes( block.name ) ) {
+			return computeTweetBlocks( block.innerBlocks );
+		}
+
+		return [];
+	} );
+};
+
+/**
+ * Handle sending the tweet refresh request.
+ *
+ * @returns {object} Refresh tweets results action.
+ */
+async function __refreshTweets() {
+	const topBlocks = select( 'core/editor' ).getBlocks();
+
+	const tweetBlocks = computeTweetBlocks( topBlocks );
+
+	try {
+		const results = await apiFetch( {
+			path: '/wpcom/v2/tweetstorm/parse',
+			data: {
+				blocks: tweetBlocks.map( block => ( {
+					attributes: block.attributes,
+					block: serialize( block ),
+					clientId: block.clientId,
+				} ) ),
+			},
+			method: 'POST',
+		} );
+
+		// Start generating any missing Twitter cards.
+		const urls = flatMap( results, block => block.urls );
+		dispatch( 'jetpack/publicize' ).getTwitterCards( urls );
+
+		return dispatch( 'jetpack/publicize' ).setTweets( results );
+	} catch ( error ) {
+		// Refreshing tweets failed
+	}
+}
+
+/**
+ * Effect handler which will refreshing the state of the tweets. Tweet refreshes are throttled
```

Comment:
Nitpick: Just a typo.

```suggestion
 * Effect handler which will refresh the state of the tweets. Tweet refreshes are throttled
```